gammapy/estimators/points/core.py

```diff
@@ -740,3 +788,127 @@ def recompute_ul(self, n_sigma_ul=2, **kwargs):
             )
         flux_points.meta["n_sigma_ul"] = n_sigma_ul
         return flux_points
+
+    def get_rebinned_axis(self, method, value, axis_name="energy"):
+        """Get the rebinned axis for rebinning
+         the flux point object along the mentioned axis.
+        The likelihoods in each bin are combined to compute the
+        resultant maps.
+
+        Parameters
+        ----------
+        method : {"fixed_bins", "fixed_edges", "min_significance"}
+            The binning method requested. Options are
+            fixed_bins : Combine `value` adjacent bins together
+            fixed_edges : Combine bins within the given edges
+            min_significance : Combine bins to get a minimum significance in each bin
+        value : The value corresponding to the method
+            The relevant matches are
+            fixed_bins : int
+            fixed_edges : MapAxis edges for the relevant axis
+            min_sig : float
+                Minimum TS desired
+        axis_name : The axis name to combine along
+
+        Returns
+        -------
+        flux_points : `~gammapy.estimators.FluxPoints`
+            A new FluxPoints object with modified binning.
+        """
+        # TODO: Make fixed_bins and fixed_edges work for multidimensions
+        if not self.geom.axes.is_unidimensional:
+            raise ValueError(
+                "Rebinning is supported only for Unidimensional FluxPoints \n "
+                "Please use `iter_by_axis` to create Unidimensional FluxPoints"
+            )
+        ax = self.geom.axes[axis_name]
+        nbin = ax.nbin
+
+        if method == "fixed_bins":
```

Comment:
I think you will arrive at a clean implementation if you use a "registry" instead: for each method you move the code to and independent function, such as `get_edges_fixed_bins()`, `get_edges_min_sqrt_ts()` etc. The you access the method like:

```python3

RESAMPLE_METHODS = {
    "fixed-bins": get_edges_fixed_bins,
    "min-sqrt-ts": get_edges_min_sqrt_ts,
    ...
}

func = RESAMPLE_METHODS[method](**kwargs)
```

This is much more extensible if future and users can even add their own methods if needed.